lib/puppet/node/bolt_project.rb

```diff
@@ -0,0 +1,46 @@
+# A project is a special type of environment used for developing Bolt content. 
+class Puppet::Node::BoltProject < Puppet::Node::Environment
+  # Return all modules for the project in the order they appear in the
+  # modulepath.
+  # @note If multiple modules with the same name are present they will
+  #   both be added, but methods like {#module} and {#module_by_forge_name}
+  #   will return the first matching entry in this list.
+  # @note This value is cached so that the filesystem doesn't have to be
+  #   re-enumerated every time this method is invoked, since that
+  #   enumeration could be a costly operation and this method is called
+  #   frequently. The cache expiry is determined by `Puppet[:filetimeout]`.
+  # @api public
+  # @return [Array<Puppet::Module>] All modules for this environment
+  def modules 
+    if @modules.nil?
+      module_references = []
+      seen_modules = {}
+
+      # This cannot have a duplicate name with boltlib modules, so it's safe to
+      # put it at the front of the modulepath
+      project_name = File.basename(Dir.pwd)
```

Comment:
We want to avoid loading ruby code from the current working directory, otherwise an unprivileged user can drop code in a directory that may be later loaded by a privileged user, leading to privilege escalation. We had one case of that in PUP-2478.

Maybe create a new puppet setting for the bolt project name, have bolt pass in the absolute path when it initializes puppet/pal, and add that path here?

Reply:
I realized I have to do this anyway since users can configure where their bolt directory is. Code is updated, I'll update the commit message after lunch!

Reply:
Thanks Josh - updated!